src/TurbulenceClosures/turbulence_closure_implementations/scalar_diffusivity.jl

```diff
@@ -219,3 +220,25 @@ function on_architecture(to, closure::ScalarDiffusivity{TD, F, <:Any, <:Any, N})
     κ = on_architecture(to, closure.κ)
     return ScalarDiffusivity{TD, F, N}(ν, κ)
 end
+
+# We do not have at the moment the ability to distinguish between the halos required in the x, y and z
+# direction for a general closure since the closure could contain any user-defined function
```

Comment:
I don't really understand this comment. We can easily tell if a closure has a function in it or not?